File: src/main/java/com/matchFit/post/dto/response/GetPost.java
package com.matchFit.post.dto.response;

import com.matchFit.post.entity.Post;
import java.util.Collections;
import java.util.List;
import java.util.Map;
import java.util.Objects;
import java.util.Set;
import java.util.stream.Collectors;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class GetPost {
    private final Long id;
    private final String title;
    private final String sports;
    private final String date;
    private final String status;
    private final String town;
    private final int cost;
    private final int currentPeople;
    private final int maxPeople;
    private final String gender;
    private final long viewCount;
    private final boolean isFollowed;

    public static List<GetPost> of(
            List<Post> posts,
            Map<Long, Long> viewCounts,
            Map<Long, Integer> currentPeopleMap,
            Set<Long> followedPostIds
    ) {
        Set<Long> followSet = followedPostIds == null ? Collections.emptySet() : followedPostIds;
        return posts.stream()
                .map(post -> {
                    Long postId = Objects.requireNonNull(post.getId());
                    return new GetPost(
                            postId,
                            post.getTitle(),
                            post.getSports().getLabel(),
                            post.getDate().toString(),
                            post.getStatus().getLabel(),
                            post.getTown().getLabel(),
                            post.getCost(),
                            currentPeopleMap.getOrDefault(postId, 0),
                            post.getMaxPeople(),
                            post.getGender().getLabel(),
                            viewCounts.getOrDefault(postId, 0L),
                            followSet.contains(postId)
                    );
                })
                .collect(Collectors.toList());
    }

    public static List<GetPost> of(
            List<Post> posts,
            Map<Long, Long> viewCounts,
            Map<Long, Integer> currentPeopleMap
    ) {
        return of(posts, viewCounts, currentPeopleMap, Collections.emptySet());
    }
}

File: src/main/kotlin/com/matchFit/post/dto/response/GetPost.kt
package com.matchFit.post.dto.response

import com.matchFit.post.entity.Post


data class GetPost(
    val id: Long,
    val title: String,
    val sports: String,
    val date: String,
    val status: String,
    val town: String,
    val cost: Int,
    val currentPeople: Int,
    val maxPeople: Int,
    val gender: String,
    val viewCount: Long
) {
    companion object {
        fun from(
            posts: List<Post>,
            viewCounts: Map<Long, Long>,
            currentPeopleMap: Map<Long, Int>
        ): List<GetPost> =
            posts.map { post ->
                val postId = requireNotNull(post.id)
                GetPost(
                    postId,
                    post.title,
                    post.sports.label,
                    post.date.toString(),
                    post.status.label,
                    post.town.label,
                    post.cost,
                    currentPeopleMap[postId] ?: 0,
                    post.maxPeople,
                    post.gender.label,
                    viewCounts[postId] ?: 0L
                )
            }
    }
}
